src/callbacks_step/analysis.jl

```diff
@@ -323,8 +323,8 @@ function (analysis_callback::AnalysisCallback)(u_ode, du_ode, integrator, semi)
             io = open(joinpath(analysis_callback.output_directory,
                                analysis_callback.analysis_filename), "a")
             @printf(io, "% 9d", iter)
-            @printf(io, "  %10.8e", t)
-            @printf(io, "  %10.8e", dt)
+            @printf(io, "  %10.15e", t)
+            @printf(io, "  %10.15e", dt)
```

Comment:
What if we used the default `print` precision? For basic floating point types, Julia will print the shortest representation allowing to recover the value.

```julia
julia> using Quadmath, DoubleFloats

julia> print(parse(Float16, "1.23456789012345678901234567890"))
1.234
julia> print(parse(Float32, "1.23456789012345678901234567890"))
1.2345679
julia> print(parse(Float64, "1.23456789012345678901234567890"))
1.2345678901234567
julia> print(parse(Float128, "1.23456789012345678901234567890"))
1.23456789012345678901234567890000003e+00
julia> print(parse(Double64, "1.23456789012345678901234567890"))
1.23456789012345678901234567889999676
```

Reply:
I am definitely in favour of this! If we want to stick with the nice formatting of the first row which prints the names of the quantities we would then need to make that also dynamic. IMO, it would also be fine to have it nice just for `Float64` and have for the other types maybe a little too few/many whitespaces.

Reply:
True :+1: